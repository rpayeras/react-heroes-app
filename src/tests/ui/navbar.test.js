import { mount } from "enzyme"
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import { Navbar } from "../../components/ui/Navbar"
import { types } from "../../types/types";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}));

const context = {
    user: {
        name: 'User',
        logged:true
    },
    dispatch: jest.fn()
}

describe('Testing navbar', () => {
    const wrapper = mount(
        <AuthContext.Provider value={context}>
            <MemoryRouter initialEntries={['/']}>
                <Navbar />
            </MemoryRouter>
        </AuthContext.Provider>
    )

    test('should render propery', () => {
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.text-info').text()).toBe('User')
    })

    test('should call logout, call the navigate and dispatch', () => {
        wrapper.find('.nav-link.btn').prop('onClick')()

        expect(context.dispatch).toHaveBeenCalledWith({
            type: types.logout
        })

        expect(mockNavigate).toHaveBeenCalledWith('/login', {
            replace: true
        })
    })
})