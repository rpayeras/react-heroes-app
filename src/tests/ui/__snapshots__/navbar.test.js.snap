// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing navbar should render propery 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Associations
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        className="nav-item nav-link"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="nav-item nav-link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="nav-item nav-link"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        User
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
